src/main/java/com/google/api/codegen/config/GapicInterfaceConfig.java

```diff
@@ -217,50 +220,17 @@ private static SmokeTestConfig createSmokeTestConfig(
       DiagCollector diagCollector,
       TargetLanguage language,
       String defaultPackageName,
-      InterfaceConfigProto interfaceConfigProto,
-      Interface apiInterface,
+      Map<Method, MethodConfigProto> methodsToGenerate,
       ResourceNameMessageConfigs messageConfigs,
       ImmutableMap<String, ResourceNameConfig> resourceNameConfigs,
       RetryCodesConfig retryCodesConfig,
       ImmutableSet<String> retryParamsConfigNames,
       ProtoParser protoParser) {
-    Map<String, GapicMethodConfig> methodConfigMapBuilder = new TreeMap<>();
```

Comment:
This structure, which contains the ordered list of methods, was accidentally a TreeMap (alphabetically ordered by name) instead of a LinkedHashMap (insertion order).

Fixing that, the method order (which follows the declarations in the protofile) is now as expected.